packages/lodestar/src/network/gossip/gossipsub.ts

```diff
@@ -89,15 +49,67 @@ export class LodestarGossipsub extends Gossipsub {
+    for(let subnet = 0; subnet < ATTESTATION_SUBNET_COUNT; subnet++) {
+      if (topic === getGossipTopic(GossipEvent.ATTESTATION_SUBNET, "ssz", new Map([["subnet", String(subnet)]]))) {
+        return this.validator.isValidIncomingCommitteeAttestation as GossipMessageValidatorFn;
+      }
+    }
+
+    switch(topic) {
+      case getGossipTopic(GossipEvent.BLOCK, "ssz"):
+        return this.validator.isValidIncomingBlock as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.ATTESTATION, "ssz"):
+        return this.validator.isValidIncomingUnaggregatedAttestation as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.AGGREGATE_AND_PROOF, "ssz"):
+        return this.validator.isValidIncomingAggregateAndProof as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.ATTESTER_SLASHING, "ssz"):
+        return this.validator.isValidIncomingAttesterSlashing as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.PROPOSER_SLASHING, "ssz"):
+        return this.validator.isValidIncomingProposerSlashing as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.VOLUNTARY_EXIT, "ssz"):
+        return this.validator.isValidIncomingVoluntaryExit as GossipMessageValidatorFn;
+      default:
+        throw new Error(`no validator for topic ${topic}`); 
+    }
+  }
+
+  private getGossipObject(topic: string, message: IGossipMessage): { object: GossipObject; subnet?: number} {
+    for(let subnet = 0; subnet < ATTESTATION_SUBNET_COUNT; subnet++) {
+      if (topic === getGossipTopic(GossipEvent.ATTESTATION_SUBNET, "ssz", new Map([["subnet", String(subnet)]]))) {
+        return {object: deserializeGossipMessage<Attestation>(message, this.config.types.Attestation), subnet};
+      }
+    }
+    switch(topic) {
+      case getGossipTopic(GossipEvent.BLOCK, "ssz"):
+        return {object: deserializeGossipMessage<SignedBeaconBlock>(message, this.config.types.SignedBeaconBlock)};
+      case getGossipTopic(GossipEvent.ATTESTATION, "ssz"):
+        return {object: deserializeGossipMessage<Attestation>(message, this.config.types.Attestation)};
+      case getGossipTopic(GossipEvent.AGGREGATE_AND_PROOF, "ssz"):
+        return {object: deserializeGossipMessage<AggregateAndProof>(message, 
+          this.config.types.AggregateAndProof)};
+      case getGossipTopic(GossipEvent.ATTESTER_SLASHING, "ssz"):
+        return {object: deserializeGossipMessage<AttesterSlashing>(message, this.config.types.AttesterSlashing)};
+      case getGossipTopic(GossipEvent.PROPOSER_SLASHING, "ssz"):
+        return {object: deserializeGossipMessage<ProposerSlashing>(message, this.config.types.ProposerSlashing)};
+      case getGossipTopic(GossipEvent.VOLUNTARY_EXIT, "ssz"):
+        return {object: deserializeGossipMessage<SignedVoluntaryExit>(message,
+          this.config.types.SignedVoluntaryExit)};
+      default:
+        throw new Error(`cannot get gossip object for topic ${topic}`); 
```

Comment:
Maybe "Don't know how to deserialize object received under topic ${topic}" would be more appropriate